### PR TITLE
Add `HighestSlotBlocksBelow` getter for db 

### PR DESCRIPTION
**Pull request by terencechain**
Review and merge #5194 first

Added `HighestSlotBlocksBelow` to retrieve blocks with the highest slot below an input slot

Change list:
* added db getter `HighestSlotBlocksBelow`
* refactored `blocksAtSlotBitfieldIndex` to be shared between `HighestSlotBlocks` and `HighestSlotBlocksBelow`

### CODE REVIEW

File: beacon-chain/db/kv/blocks.go [hunk 1]
```diff
@@ -315,31 +316,67 @@ func (k *Store) HighestSlotBlock(ctx context.Context) (*ethpb.SignedBeaconBlock,
...
+	highestSlot = uint64(math.Max(0, float64(highestSlot)))
+	f := filters.NewFilter().SetStartSlot(highestSlot).SetEndSlot(highestSlot)
+
+	keys, err := getBlockRootsByFilter(ctx, tx, f)
+	if err != nil {
+		return nil, err
 	}
 
-	return blocks[0], err
+	blocks := make([]*ethpb.SignedBeaconBlock, 0)
```
Comment (0xKiwi) [inline beacon-chain/db/kv/blocks.go]: Can you declare the length of this?

File: beacon-chain/db/kv/blocks.go [hunk 2]
```diff
@@ -315,31 +316,67 @@ func (k *Store) HighestSlotBlock(ctx context.Context) (*ethpb.SignedBeaconBlock,
...
+	return blocks, err
+}
+
+// blocksAtSlotBitfieldIndex retrieves the block in DB given the input index. The index represents
+// the position of the slot bitfield the saved block maps to.
+func (k *Store) blocksAtSlotBitfieldIndex(ctx context.Context, tx *bolt.Tx, index uint64) ([]*ethpb.SignedBeaconBlock, error) {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.blocksAtSlotBitfieldIndex")
+	defer span.End()
+
+	highestSlot := index - 1
```
Comment (0xKiwi) [inline beacon-chain/db/kv/blocks.go]: index can never be 0?

Comment (terencechain) [reply beacon-chain/db/kv/blocks.go]: The underflow check is below this line

File: beacon-chain/db/kv/blocks.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/db/kv/blocks.go]: Good point

File: beacon-chain/db/kv/blocks.go [hunk 3]
```diff
@@ -316,30 +316,69 @@ func (k *Store) HighestSlotBlocks(ctx context.Context) ([]*ethpb.SignedBeaconBlo
...
 
-		keys, err := getBlockRootsByFilter(ctx, tx, f)
+		blocks, err = k.blocksAtSlotBitfieldIndex(ctx, tx, uint64(highestIndex))
 		if err != nil {
 			return err
 		}
+		return nil
+	})
 
-		bBkt := tx.Bucket(blocksBucket)
```
Comment (terencechain) [inline beacon-chain/db/kv/blocks.go]: refactored this into `blocksAtSlotBitfieldIndex`

**Pull request merged**